server/src/config.ts

```diff
@@ -361,7 +361,7 @@ export const immichAppConfig: ConfigModuleOptions = {
   envFilePath: '.env',
   isGlobal: true,
   validationSchema: Joi.object({
-    IMMICH_ENV: Joi.string().optional().valid('development', 'production').default('production'),
+    IMMICH_ENV: Joi.string().optional().valid('development', 'e2e', 'production').default('production'),
```

Comment:
nit: everything is spelled out and e2e is abbreviated. I'd be fine with `testing`